### ISSUE TITLE
Model run through `CPUExecutionProvider` and `CUDAExecutionProvider` do not yield the same output

### ISSUE THREAD

**Issue by fxmarty** [microsoft/onnxruntime#14500]
### Describe the issue

Hello,

I exported a stable diffusion model to ONNX as a single file with multiple subgraphs, that contain a `Loop` node, and several `If` nodes.

The model runs fine through `CUDAExecutionProvider`:

![image](https://user-images.githubusercontent.com/9808326/215756918-03371fc1-e0cd-403a-b427-fb8275ce819f.png)

The model runs through `CPUExecutionProvider` yields garbage:

![image](https://user-images.githubusercontent.com/9808326/215756968-496b4fa1-d438-453e-885b-02c17d1eaf83.png)

This is consistent across several runs.

### To reproduce

Refer to this model: https://huggingface.co/fxmarty/end-to-end-stable-diffusion-onnx/tree/main (beware that the external data was not uploaded, so this is **just for inspection purpose**), it is exported from `CompVis/stable-diffusion-v1-4`. Let me know if you need the full model.

Have a look at netron: https://netron.app/?url=https://huggingface.co/fxmarty/end-to-end-stable-diffusion-onnx/blob/main/stable_diffusion_pipeline.onnx

You can refer to the code used to generate this model: https://github.com/huggingface/optimum/pull/730

The script used for inference (the original prompt is "A cat sleeping on the beach"):

```python
import onnxruntime as ort
import numpy as np
import torch

def numpy_to_pil(images):
    """
    Convert a numpy image or a batch of images to a PIL image.
    """
    if images.ndim == 3:
        images = images[None, ...]
    images = (images * 255).round().astype("uint8")
    if images.shape[-1] == 1:
        # special case for grayscale (single channel) images
        pil_images = [Image.fromarray(image.squeeze(), mode="L") for image in images]
    else:
        pil_images = [Image.fromarray(image) for image in images]

    return pil_images

# replace here to CUDAExecutionProvider if needed
session = ort.InferenceSession("stable_diffusion_pipeline.onnx", providers=["CPUExecutionProvider"])

text_input_ids = torch.tensor([[49406,   320,  2368,  6982,   525,   518,  2117, 49407, 49407, 49407,
         49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
         49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
         49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
         49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
         49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
         49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
         49407, 49407, 49407, 49407, 49407, 49407, 49407]])
uncond_text_input_ids = torch.tensor([[49406, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
         49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
         49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
         49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
         49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
         49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
         49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407, 49407,
         49407, 49407, 49407, 49407, 49407, 49407, 49407]])
timesteps = torch.tensor([981, 961, 961, 941, 921, 901, 881, 861, 841, 821, 801, 781, 761, 741,
        721, 701, 681, 661, 641, 621, 601, 581, 561, 541, 521, 501, 481, 461,
        441, 421, 401, 381, 361, 341, 321, 301, 281, 261, 241, 221, 201, 181,
        161, 141, 121, 101,  81,  61,  41,  21,   1])

inp = {
    "text_input_ids": text_input_ids.numpy().astype(np.int64),
    "uncond_text_input_ids": uncond_text_input_ids.numpy().astype(np.int64),
    "timesteps": timesteps.numpy().astype(np.int64),
}

np_image = session.run(None, inp)[0]

image = numpy_to_pil(np_image)
image[0].save("ort_out.png")
```
The scripts used to obtain the model can be found at https://github.com/huggingface/optimum/pull/730 . The TL;DR of it is:

1. `torch.jit.trace` the text encoder, unet and VAE decoder
2. `torch.jit.script` the whole pipeline, where the text encoder, unet and VAE decoder are already ScriptModules so not re-scripted.
3. Export with `torch.onnx.export` the obtained ScriptModule.

### Urgency

It's likely an issue in an operator implementation in CPUExecutionProvider, so I'd say rather high.

### Platform

Linux

### OS Version

Linux 5.15.0-56-generic #62-Ubuntu SMP Tue Nov 22 19:54:14 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.13.1

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

Default CPU

**Comment by fxmarty** [comment]
Note that running the `ScriptModule` on CPU gives good results. You can checkout on https://github.com/huggingface/optimum/pull/730, run `python create_scriptmodule.py` and then `python run_scriptmodule_on_cpu.py` to verify.

**Comment by fxmarty** [comment]
@tianleiwu Thank you for your help offline. Using a version of ONNX Runtime built from source with `ORT_DEBUG_NODE_IO_DUMP_OUTPUT_DATA=1`, I pinpoint the issue to:

On CPU:
![image](https://user-images.githubusercontent.com/9808326/216053128-5cfaf32f-55e2-4ddf-a068-66b653f70997.png)

On GPU:
![image](https://user-images.githubusercontent.com/9808326/216053274-3c92e103-0f8f-4fa9-8ec7-433ba80d1387.png)

Note that inspecting the log, the input `/unet/0/0_1/Transpose_1_output_0` is identical on `CPUExecutionProvider` and `CUDAExecutionProvider`.

Hence, it seems that the issue comes from `FusedConv`.

**Comment by fxmarty** [comment]
@tianleiwu Thank you for your help offline. Using a version of ONNX Runtime built from source with `ORT_DEBUG_NODE_IO_DUMP_OUTPUT_DATA=1`, I pinpoint the issue to:

On CPU, a single `FusedConv` is used:
![image](https://user-images.githubusercontent.com/9808326/216053128-5cfaf32f-55e2-4ddf-a068-66b653f70997.png)

On GPU, two nodes `Conv` and `Add` are used:
![image](https://user-images.githubusercontent.com/9808326/216053274-3c92e103-0f8f-4fa9-8ec7-433ba80d1387.png)

![image](https://user-images.githubusercontent.com/9808326/216054123-5c3d8d5d-9f30-4d83-a662-a24fbe7fdb3f.png)

We see that the output `/unet/0/0_1/Add_output_0` differs between the two. Note that inspecting the log, the input `/unet/0/0_1/Transpose_1_output_0` is identical on `CPUExecutionProvider` and `CUDAExecutionProvider`.

Hence, it seems that the issue comes from `FusedConv` on CPU.

**Comment by fxmarty** [comment]
Can confirm the issue lies in using `GraphOptimizationLevel.ORT_ENABLE_EXTENDED`. There is no issue using `GraphOptimizationLevel.ORT_ENABLE_BASIC`.

**Comment by wejoncy** [comment]
Hi,
Could you please share the two tensor's shape?
![image](https://user-images.githubusercontent.com/9417365/216335628-1009dea5-ca97-463e-aa08-cbe931be33c7.png)

And the attributes of the Conv node also be a key info.
![image](https://user-images.githubusercontent.com/9417365/216335777-f2506640-603f-40c2-b8c4-2c05181db725.png)
 
Thanks

**Comment by fxmarty** [comment]
>  Could you please share the two tensor's shape?

Hi, for sure! I would need to run with `ORT_DEBUG_NODE_IO_DUMP_SHAPE_DATA=1`, right?

> And the attributes of the Conv node also be a key info

What do you mean by attributes?

**Comment by wejoncy** [comment]
1), Yeah. Thanks

2). It's the padding/strides/kernel size, etc. I have had it from the netron tool. It just took me a long time to load the subgraph.

**Comment by wejoncy** [comment]
It's quite weired.

I am wondering if the input tensor in CPU EP was all zeros?
`Input 3 Name: /unet/0/0/Div_output_0`

Can we disscuss it offline if you are urgent to tackle it? I may need more info to understant where the error is from.

**Comment by fxmarty** [comment]
Thank you for your help, for sure, let me know how I can reach you offline.

What is surprising is that the output of the `FusedConv` on CPU EP and `Conv` on CUDA EP are identical, only that the CUDA EP has the `Add` node on top (that is probably eaten with optimization level 99). Would it help to pass a session option `optimized_model_filepath` to save the optimized graph and share it with you?

I believe the operator corresponds to: https://github.com/huggingface/diffusers/blob/2bbd532990643a2aea5f1b33ec2fe3d6e7def4b1/src/diffusers/models/transformer_2d.py#L205

We indeed have the attributes:
![image](https://user-images.githubusercontent.com/9808326/216541157-58cd12a7-8a93-4a0a-a294-51f0cc958fc4.png)


The `/unet/0/0/Div_output_0` is identical on both CUDA and CPU EP:
![image](https://user-images.githubusercontent.com/9808326/216541680-6ef2dca9-d4ff-4159-bebe-c33398ff3ab7.png)

![image](https://user-images.githubusercontent.com/9808326/216541718-f5c78a56-503b-489a-ab75-d1fa39c7be2b.png)

**Comment by wejoncy** [comment]
Would you mind sharing me the FULL MODEL so I can reproduce it in local? 
Please Contact me (jicwen@microsoft.com) if it's doable.

**Comment by fxmarty** [comment]
For sure, you can use:
```
wget https://felix-playground.s3.eu-west-3.amazonaws.com/onnx_model.zip
wget https://felix-playground.s3.eu-west-3.amazonaws.com/stable_diffusion_pipeline.onnx_data
```

The zip file contains a few external data and the .onnx file.

**Comment by wejoncy** [comment]
> For sure, you can use:
> 
> ```
> wget https://felix-playground.s3.eu-west-3.amazonaws.com/onnx_model.zip
> wget https://felix-playground.s3.eu-west-3.amazonaws.com/stable_diffusion_pipeline.onnx_data
> ```
> 
> The zip file contains a few external data and the .onnx file.
> 
> On CUDA EP, note that it takes quite a bit of memory, >12 GB.

Not sure why I can't reproduce it. I can always get the correct result with CPU EP. Do you have any special confs for ORT? or could you share you CPU info? say "Intel(R) Core(TM) i9-10900X CPU @ 3.70GHz"

**Comment by fxmarty** [comment]
Thank you! I tried out with `1.14.0.dev20230119003`, it appears the issue has been fixed in the nightlies!

**Comment by wejoncy** [comment]
Great. I'll close the issue then.
